proto-compiler/shell.nix

```diff
@@ -0,0 +1,12 @@
+# This file exists for legacy nix-shell
+# https://nixos.wiki/wiki/Flakes#Using_flakes_project_from_a_legacy_Nix
```

Comment:
Since this is a new project, would it be better if we just use Nix Flakes from the get go and forget about legacy Nix? I understand that Nix Flakes is still experimental, but we currently do not have the baggage of needing to support legacy Nix either.